src/test/ui/async-await/async-fn-nonsend.stderr

```diff
@@ -72,6 +77,12 @@ LL |     assert_send(non_sync_with_method_call());
    |     ^^^^^^^^^^^ future returned by `non_sync_with_method_call` is not `Send`
    |
    = help: within `std::fmt::ArgumentV1<'_>`, the trait `std::marker::Sync` is not implemented for `*mut (dyn std::ops::Fn() + 'static)`
+   = note: required because it appears within the type `std::marker::PhantomData<*mut (dyn std::ops::Fn() + 'static)>`
```

Comment:
I have to say that I'm not sure how much value this provides either. Based on the error message, at least, I admit I can't really make heads or tails of these types. At best, it seems sort of like they're appearing in the wrong place, as they seem to be explaining something about the variable `f` below (at least that's the only mention of formatting that I see).

I guess I feel like in practice if I saw this error I'd just skip over all those notes and I'd probably tell users to do the same.

That said, I can imagine that in some very confusing cases notes like this could be helpful -- but I'd sort of like an example of such a case.